modules/b2b_logic/logic.c

```diff
@@ -1123,15 +1180,26 @@ int _b2b_handle_reply(struct sip_msg *msg, b2bl_tuple_t *tuple,
 	if (tuple->state == B2B_BRIDGING_STATE) {
 		LM_DBG("Received a reply [%d] while in BRIDGING scenario\n",
 			statuscode);
+
 		/* if the scenario state is B2B_BRIDGING_STATE -> we should have a reply for INVITE */
 		/* extract the method from Cseq header */
 
+		if(method_value == METHOD_NOTIFY) goto done1; /* Silently ignore reply on NOTIFY */
+
 		if(method_value != METHOD_INVITE)
 		{
 			LM_ERR("Wrong scenario state [B2B_BRIDGING_STATE] for this"
 				" reply(for method %d)\n", method_value);
 			goto error;
 		}
+
+		if(statuscode >= 200 && entity == tuple->bridge_entities[1]) /* Reply from new bridge entity */
+		{
+			process_bridge_notify(tuple->servers[0], cur_route_ctx.hash_index, msg);
+			if(statuscode == 200 || !(tuple->bridge_flags & B2BL_BR_FLAG_RETURN_AFTER_FAILURE))
+				b2bl_delete_entity(tuple->servers[0], tuple, tuple->hash_index, 1);
```

Comment:
Following the same logic as above, I think we should not delete this entity automatically, if we are not in a refer scenario.